src/fev/task.py

```diff
@@ -256,68 +366,115 @@ def __post_init__(self):
+    @property
+    def cutoffs(self) -> list[int] | list[str]:
+        """List of cutoffs for each evaluation window."""
+        if isinstance(self.initial_cutoff, int):
+            assert isinstance(self.window_step_size, int)
+            return [self.initial_cutoff + window_idx * self.window_step_size for window_idx in range(self.num_windows)]
+        else:
+            assert isinstance(self.initial_cutoff, str)
+            if isinstance(self.window_step_size, str):
+                offset = pd.tseries.frequencies.to_offset(self.window_step_size)
+            else:
+                assert isinstance(self.window_step_size, int)
+                offset = pd.tseries.frequencies.to_offset(self.freq) * self.window_step_size
+
+            cutoffs = []
+            for window_idx in range(self.num_windows):
+                cutoff = pd.Timestamp(self.initial_cutoff)
+                # We don't add the offset for window_idx=0 to avoid applying an "anchored" offset
+                # (e.g. `Timestamp("2020-01-01") + i * to_offset("ME")` returns "2020-01-31" for i=0 and i=1)
+                if window_idx != 0:
+                    cutoff += window_idx * offset
+                cutoffs.append(cutoff.isoformat())
+            return cutoffs
 
-            Contains columns corresponding to `task.id_column`, `task.timestamp_column`, `task.static_columns` and
-            `task.dynamic_columns` (except those in `task.past_dynamic_columns`).
-        """
-        if self._dataset_dict is None:
-            self._prepare_dataset_dict(
-                num_proc=num_proc, storage_options=storage_options, trust_remote_code=trust_remote_code
-            )
-        return self._dataset_dict[TRAIN], self._dataset_dict[FUTURE]
+    def to_dict(self) -> dict:
+        """Convert task definition to a dictionary."""
+        return dataclasses.asdict(self)
 
-    def get_test_data(
+    def load_full_dataset(
         self,
-        num_proc: int = DEFAULT_NUM_PROC,
         storage_options: dict | None = None,
         trust_remote_code: bool | None = None,
+        num_proc: int = DEFAULT_NUM_PROC,
     ) -> datasets.Dataset:
-        """Get the future data that must be predicted.
-
-        This data is only used for evaluating forecast accuracy and should not be passed to the model.
-        """
-        if self._dataset_dict is None:
-            self._prepare_dataset_dict(
-                num_proc=num_proc, storage_options=storage_options, trust_remote_code=trust_remote_code
+        """Load the full raw dataset. For model evaluation, use iter_windows() instead."""
+        if self._full_dataset is None:
+            self._full_dataset = self._load_dataset(
+                storage_options=storage_options, trust_remote_code=trust_remote_code, num_proc=num_proc
             )
-        return self._dataset_dict[TEST]
+        return self._full_dataset
+
+    def iter_windows(self, **load_full_dataset_kwargs) -> Iterable[EvaluationWindow]:
```

Comment:
I would spell out the kwargs _and_ document them, as this looks like it will be an often used method of Task. Currently, every time I use it vscode will tell me it accepts some kwargs, it doesn't know what those kwargs are. I should also understand the library has broken from `kwargs` convention probably for some meaningful reason so I should google stuff.

Same for `get_window`.